README.mkd

```diff
@@ -229,6 +229,57 @@ registers and then save `ra` and the appropriate number of registers from
 epilogue. They restore the saved registers, deallocate the stack space for the
 register, and then perform a return through the restored value of `ra`.
 
+## Conventions for vendor extension
+
+Support for custom instruction set extensions are an important part of RISC-V,
+with large encoding spaces reserved of vendor extensions.
+
+However, there are no official guidelines on naming the mnemonics. This section
+defines guidelines which vendors are expected to follow if upstreaming support
+for their extensions. Although vendor-provided toolchains are free to make
+different choices, they are strongly urged to align with these guidelines in
+order to ensure there is a straightforward path for upstreaming in the future.
+
+### Vendor extension naming scheme
+
+According to the RISC-V ISA spec, non-standard extensions are named using a single `x`
+followed by an alphabetical name and an optional version number.
+
+To make it easier to identify and prevent naming conflict, vendor extensions must start
+with a vendor name, which could be an abbreviation of the full name.
+
+For example:
+- `XVentanaCondOps` from Ventana
+- `Xsfcflushdlone` from SiFive
+- `Xtheade` from T-head
+
+### Assembly mnemonic
+
+In order to avoid confusion between standard extension and other vendor
+extensions, instruction mnemonics from vendor extensions must have a
+prefix corresponding to the vendor's name.
+
+e.g. `sf.` for SiFive, `vt.` for Ventana. No central registration with RISC-V
+International or elsewhere is required before the prefix is used.
```

Comment:
Please add a comment that, although no centralized registration is required, they should add the vendor prefix to the table below, with a link to the section.

Reply:
Note added :)